csrc/device_lower/pass/insert_syncs.cpp

```diff
@@ -587,10 +587,18 @@ class ReadAfterWriteSyncs : public kir::ExprMutator {
 
       if (!sync_bitmap.hasBID() &&
           std::all_of(
-              expr->inputs().begin(), expr->inputs().end(), [](Val* val) {
+              expr->inputs().begin(),
+              expr->inputs().end(),
+              [](Val* val) {
                 return !val->isA<TensorView>() ||
                     !isSharedMemory(val->as<TensorView>()) ||
                     ir_utils::isCpAsyncBulkLoad(val->definition());
+              }) &&
```

Comment:
This is a temporary WAR for #4741. Not sure if this is appropriate. Needs help from @zasdfgbnm